napalm_brocade/brocade.py

```diff
@@ -0,0 +1,428 @@
+
+        cmd = "copy scp://%s:%s@%s/tmp/%s flash://_candidate.cfg" \
+            % (EXPORT_USER, EXPORT_PASSWORD, EXPORT_HOST, filename)
+
+        self.device.send_command(cmd)
+
+    def load_merge_candidate(self, filename=None, config=None):
+
+        dst = "%s/tmp/%s" % (os.environ['HOME'], filename)
+        copyfile(filename, dst)
+        cmd = "oscmd rm /var/config/vcs/scripts/_candidate.cfg"
+        self.device.send_command(cmd)
+        cmd = "copy scp://%s:%s@%s/tmp/%s flash://_candidate.cfg" \
+            % (EXPORT_USER, EXPORT_PASSWORD, EXPORT_HOST, filename)
+
+        self.device.send_command(cmd)
+
+    def rollback_config(self):
+
+        print "Reloading previous checkpoint ..."
+        cmd = "copy flash://_running.cfg running_config"
+        self.device.send_command(cmd) 
+        self.reboot()
+
+    def compare_config(self):
+
+        cmd = "oscmd diff /var/config/vcs/scripts/_running.cfg /var/config/vcs/scripts/_candidate.cfg"
+        self.device.send_command(cmd)
+
+    def discard_config(self):
+        cmd = "oscmd rm /var/config/vcs/scripts/_candidate.cfg"
+        self.device.send_command(cmd)
+
+    def get_interfaces_counters(self):
+        cmd = "show interface stats brief"
+        lines = self.device.send_command(cmd)
+        lines = lines.split('\n')
+
+        counters_table = []
+        # Skip the first four lines which is the header
+        lines = lines[4:-1]
+
+        for line in lines:
+            if len(line) == 0:
+                return {}
+            if len(line.split()) == 9:
+                interface_type, interface, pkts_rx, pkts_tx, \
+                    err_rx, err_tx, discards_rx, discards_tx, crc_rx = \
+                    line.split()
+                entry = {
+                    'interface_type': interface_type,
+                    'interface': interface,
+                    'pkts_rx': pkts_rx,
+                    'pkts_tx': pkts_tx
+                }
+                counters_table.append(entry)
+            else:
+                raise ValueError(
+                    "Unexpected output from: {}".format(line.split()))
+
+        return counters_table
+            
+    def get_mac_address_table(self):
+        """Get mac address table (TBD)."""
+
+        cmd = "show mac-address-table"
+        lines = self.device.send_command(cmd)
+        lines = lines.split('\n')
+
+        mac_address_table = []
+        # Skip the first four lines which is the header
+        lines = lines[1:-1]
+
+        for line in lines:
+            if len(line) == 0:
+                return {}
+            if len(line.split()) == 7:
+                vlanid, tt, mac_address, type, state, port_type, port = \
+                    line.split()
+                entry = {
+                    'vlanid': vlanid,
+                    'mac+address': mac_address,
+                    'type': type,
+                    'state': state,
+                    'port_type': port_type,
```

Comment:
Field ```port_type``` should not exist: https://github.com/napalm-automation/napalm-base/blob/develop/napalm_base/test/models.py#L194